build-image/Dockerfile

```diff
@@ -23,7 +23,7 @@ RUN GOARCH=$(go env GOARCH) && \
 	chmod +x shfmt && \
 	mv shfmt /usr/bin
 
-RUN curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b /usr/bin v1.64.5
+RUN curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b /usr/bin v2.1.6
```

Comment:
Put this change in a separate PR, so you get a new build image that you can use
https://github.com/cortexproject/cortex/blob/master/.github/workflows/build-image.yml